### PR TITLE
Feature/kaa 95

### PR DESCRIPTION
**Pull request by ghost**
Logging sybsystem implementation in C SDK. C code generation in control server.

### CODE REVIEW

**Comment by ikulikov** [comment]
LGFM

File: client/client-multi/client-c/CMakeLists.txt [hunk 1]
```diff
@@ -52,11 +56,17 @@ set (KAA_SOURCE_FILES
                        src/kaa_channel_manager.c
                        src/kaa_common_schema.c
                        src/kaa_mem.c
+                       src/kaa_uuid_int.c
                        src/kaa_common.c
                        src/gen/kaa_endpoint_gen.c
                        src/gen/kaa_profile_gen.c
+                       src/gen/kaa_logging_gen.c
```
Comment (ghost) [inline client/client-multi/client-c/CMakeLists.txt]: All "kaa_*_gen.c" files (except kaa_endpoint_gen.c) should be placed in CMakeGen.cmake

File: client/client-multi/client-c/src/kaa_uuid_int.c [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+void kaa_uuid_copy(kaa_uuid_t *dst, kaa_uuid_t *src)
+{
+    *dst = *src;
+}
+
+void kaa_uuid_to_string(char **dst, kaa_uuid_t *uuid)
+{
+    char buf[16];
+    sprintf(buf, "%i", *uuid);
```
Comment (ghost) [inline client/client-multi/client-c/src/kaa_uuid_int.c]: %lu or %llu

File: client/client-multi/client-c/src/kaa_uuid_int.c [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    size_t len = strlen(buf);
```
Comment (ghost) [inline client/client-multi/client-c/src/kaa_uuid_int.c]: Better to use return value of sprintf call

File: client/client-multi/client-c/src/kaa_uuid_int.c [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    *dst = KAA_CALLOC(len + 1, sizeof(char));
+    strcpy(*dst, buf);
+}
+
+void kaa_uuid_from_string(const char *src, kaa_uuid_t *uuid)
+{
+    sscanf(src, "%i", uuid);
```
Comment (ghost) [inline client/client-multi/client-c/src/kaa_uuid_int.c]: %lu or %llu

**Comment by ghost** [comment]
LGTM

**Pull request merged**